lib/swagger/resources.js

```diff
@@ -74,9 +72,10 @@ swaggerPaths._getResourceDefinition = function(resourceConfig) {
       }
       attributeShortcut[attribute] = swaggerScheme;
 
-      // if ((joiScheme._flags || { }).presence === "required") {
-      //   resourceDefinition.properties.attributes.required.push(attribute);
-      // }
+      if ((joiScheme._flags || { }).presence === "required") {
+        resourceDefinition.properties.attributes.required = resourceDefinition.properties.attributes.required || [ ];
+        resourceDefinition.properties.attributes.required.push(attribute);
+      }
```

Comment:
This block used to give us troubles with our test suite, especially around queries that use the `fields=` parameter to opt-out of seeing required attributes. I've fixed the tests and this can go back in. Required fields now appear correctly in the swagger doc.